google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java

```diff
@@ -686,15 +697,23 @@ private void doneCallback(Throwable finalStatus) {
     try {
       this.streamConnectionIsConnected = false;
       if (connectionFinalStatus == null) {
+        if (connectionRetryStartTime == 0) {
+          connectionRetryStartTime = System.currentTimeMillis();
+        }
         // If the error can be retried, don't set it here, let it try to retry later on.
-        if (isRetriableError(finalStatus) && !userClosed) {
+        if (isRetriableError(finalStatus)
+            && !userClosed
+            && System.currentTimeMillis() - connectionRetryStartTime
+                <= maxRetryDuration.toMillis()) {
```

Comment:
What if the user doesn't want a max retry duration enforced?  Should there be special handling for the zero value?

Reply:
Sure.